packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +39,176 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
   await writeCFNTemplate(cfnTemplate, newPath, { templateFormat });
   return newPath;
 }
+
+//get data from custom polcies file and write custom policies to CFN template
+export async function writeCustomPoliciesToCFNTemplate(
+  resourceName: string,
+  service: string,
+  resourceDir: string,
+  cfnFile: string,
+  category: string,
+  filePath: string
+) {
+  if (!(service === 'Lambda' || service === 'ElasticContainer')) {
+    return;
+  }
+  const { templateFormat, cfnTemplate } = await readCFNTemplate(path.join(resourceDir, cfnFile));
+  const customPolicies = stateManager.getCustomPolicies(category, resourceName);
+  if (!validateExistCustomPolicies(customPolicies)) {
```

Comment:
whats the differecne between `validateExistCustomPolicies` and `validateCustomPoliciesSchema` function?

Reply:
validateExistCustomPolicies is to validate if customers add any custom policies, if not, we will see if exists and delete the policies in the CFN template and skip the process. validateCustomPoliciesSchema is to validate if they enter the right schema of policies, if not, we will throw an error